### PR TITLE
[docs] Install ray from conda-forge instead of PyPi when using conda

### PR DESCRIPTION
**Pull request by tupui**
## Why are these changes needed?

The installation instructions when using a conda environment are referring to PyPi. Instead it should show how to use the libraries distributed on Conda-forge.

## Related issue number

Related to #25291

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [x] This PR is not tested :(

### CODE REVIEW

**Comment by pcmoritz** [comment]
@richardliaw What do you think about this change? Is Ray on conda-forge always up to date and working well?

**Comment by tupui** [comment]
The only drawback I can see is that at the moment there is no macOS version on Conda-forge. 

Otherwise, using conda for large packages especially when they have complex dependencies and compiled code is always a better user experience with conda.

pip is getting better, but it is still fairly easy to compromise an environment. On the other hand, conda would not let you break an environment. Hence showing users how to mix pip with conda would most certainly result in failures (as seen in the linked issue) and is a bad practice.

**Comment by jmakov** [comment]
Would just like to point out that the [`ray` package on `conda froge`](https://anaconda.org/conda-forge/ray-default) is 1 version behind the latest `ray` `pip` version even though latest `ray` `pip` version was published 7 days ago. Seems like a small continuous deployment issue to me (would be great if as part of you publishing the pip package also `conda` packages would be published).

**Comment by tupui** [comment]
Normally conda is creating automatically a PR to bump the version so you "just" need to review and merge when CI is green. You could create the PR manually before to prepare things and have a faster deployment on conda (since it takes a few hours after you merge the PR).

File: doc/source/ray-overview/installation.rst
Comment (tupui) [inline doc/source/ray-overview/installation.rst:237]:
```suggestion
For a complete list of available ``ray`` libraries, have a look at: https://github.com/conda-forge/ray-packages-feedstock

.. note::

  At this time, Ray is not yet distributed on Conda-forge for macOS.
  Hence you need to use `pip install ray` in the newly created environment.

.. _`Anaconda`: https://www.anaconda.com/
```

**Comment by jjyao** [comment]
IMO we should use pypi package if possible. The conda package is not vigorously tested like the pypi package that is officially maintained by the Ray team. Also conda package is not complete as the pypi one: for example it doesn't support mac osx.

cc @krfricke

**Comment by tupui** [comment]
I understand the argument that currently it's not being tested and supported, but I would argue that it should become the case.

FYI for SciPy/NumPy, only conda provides builds with full parallelism enabled. For complex packages with build dependencies, conda is IMHO the only sane option at the moment. This is why the whole PyData stack is advocating for using it.

**Comment by jjyao** [comment]
@tupui Until we reach the fully supported stage, I think we should still recommend pypi package.

Also feel free to help us with the conda package: https://github.com/conda-forge/ray-packages-feedstock :)

**Comment by jmakov** [comment]
@jjyao I'm the original poster of https://github.com/ray-project/ray/issues/25291. Having dependencies described with `conda` the installation with `pip` didn't work. `conda`/`mamba` should become the primary way of installing `ray` since it can install also packages not in PyPi.

**Comment by jjyao** [comment]
@jmakov Got it. I think for now we should still recommend pypi package until the conda package is mature enough. But we can add a paragraph saying that if pip install is not an option, we also have the community supported conda package that people can try. WDYT? @tupui

**Comment by tupui** [comment]
`pip` is still the default install method after considering this PR. I pretty much trust the folks at conda-forge that their repackaging is working.

Still I can rephrase and add that even using a conda environment, `pip` is the recommended/supported way.

**Comment by jjyao** [comment]
@tupui Just my limited experience of releasing Ray 1.7 conda package:
1. Currently ray conda package is not built directly from the ray source code without modification, we need to provide some patches in order for it to work. Those patches are not fully tested.
2. The dependencies of Ray conda package have to be conda packages as well, some of them are not as mature as their pypi counterpart (I have encountered bugs due to that).

**Comment by tupui** [comment]
Sorry I did not realize you did conda release. Then sure if you are saying it's not stable enough to advertize for now, then I will modify the text.

**Comment by tupui** [comment]
@jjyao I made the changes recommending PyPi install.

File: doc/source/ray-overview/installation.rst
Comment (jjyao) [inline doc/source/ray-overview/installation.rst:230]: nit: Could we move the note to the end of the section?

Ray conda packages are maintained by the community, not the Ray team. While using a
  conda environment, it is recommended to install Ray from PyPi if possible.

Comment (tupui) [reply doc/source/ray-overview/installation.rst]: Done

**Review by jjyao — APPROVED** [review]

**Comment by jjyao** [comment]
@maxpumperla, @richardliaw  Could you take another look as the code owner?

**Review by richardliaw — APPROVED** [review]

**Pull request merged**
